plugins/filebrowser.c

```diff
@@ -736,13 +739,9 @@ static gboolean on_button_press(GtkWidget *widget, GdkEventButton *event, gpoint
 	}
 	else if (event->button == 3)
 	{
-		static GtkWidget *popup_menu = NULL;
```

Comment:
While I dislike static local (or global) variables which allocate memory dynamically and never cleanup, moving this one to the global scope seems completely unrelated to this change. Or did I miss it being used elsewhere?

Reply:
The old code have no sense...can you explain `if (popup_menu == NULL)`?
https://github.com/geany/geany/blob/868da11715ed86d08c1c2c4deaf1b2aa6c209df4/plugins/filebrowser.c#L739-L742

I have to revert this?

Reply:
One of the meanings in C of `static` is that it's only initialized once for the whole runtime, so the above code initializes `popup_menu` to `NULL` for the only/one-time initialization. Then the `if (popup_menu == NULL)` branch says "if this is the first time through this code, assign `popup_menu` with some runtime value and then never enter this branch again." It's not terribly good code, IMO, but it isn't wrong either.

> I have to revert this?

I'm not strictly opposed to the change, it just looks completely unrelated to the pull request and an argument could be made that if it's only used in one place, and it's going to leak memory anyway, then having the variable definition closer to the use of it is best.

Reply:
You are right, I push a commit to revert this and add space at line 691.